README.md

```diff
@@ -46,8 +46,16 @@ leasedir: /var/lib/golang-dhcpd
 
     mkdir /etc/golang-dhcpd
     cp conf.yml /etc/golang-dhcpd/conf.yaml
-    docker build -t golang-dhcpd:latest .
-    docker-compose -f docker-compose.yml up
+    docker compose up --build -d
+    docker compose logs -f
+
+**Note:** The Docker setup uses ipvlan networking for security. Update your config to:
+- Set `interfaces: [ eth0 ]` (interface name inside container may not be the same as host interface)
+- Set `myip:` to match the container's IP address
+
+**Warning:** Using `network_mode: "host"` with unprivileged users may fail
+to bind to port 67 due to Docker limitations with capabilities in host networking mode,
+so one can either use `ipvlan` mode with dhcp user or `host` mode with root user.
```

Comment:
>**Warning:** Using `network_mode: "host"` with unprivileged users may fail to bind to port 67 due to Docker limitations with capabilities in host networking mode,

hmmm can how does this break? Eg can you paste me the error you get?

Reply:
@jrgp The error was:
```
golang-dhcpd-1  | 2025/09/25 17:39:47 Failed listening: listen udp 0.0.0.0:67: bind: permission denied
```

I suspect the issue is that capabilities set in the docker‑compose file apply to the container's root user, but the container process drops privileges to an unprivileged user before starting dhcpd, so those capabilities are lost.

One can alternatively set capabilities on the dhcpd binary in the Dockerfile (so the binary retains the needed rights after the UID drop). But using `ipvlan` rather than `host` network mode is probably the safer option.